### PR TITLE
feat: view actual download chunks and fixed download actions

### PR DESCRIPTION
**Pull request by zelfroster**
The download chunks PR based on latest commit

### CODE REVIEW

**Comment by zelfroster** [comment]
@csoler This is ready for review and merge.

File: webui-src/app/files/files_util.js [hunk 1]
```diff
@@ -91,212 +91,140 @@ function calcRemainingTime(bytes, rate) {
...
-const chunkStrats = {
-  0: 'Streaming', // CHUNK_STRATEGY_STREAMING
-  1: 'Random', // CHUNK_STRATEGY_RANDOM
-  2: 'Progressive', // CHUNK_STRATEGY_PROGRESSIVE
-};
-// rstypes.h :: 366
-
 const File = () => {
   let chunkStrat;
+  let rsJsonApiRequest = () => {};
```
Comment (rottencandy) [inline webui-src/app/files/files_util.js:131]: Why is this needed?

Comment (zelfroster) [reply webui-src/app/files/files_util.js:131]: At that time I thought to initialise the function first and then assign it with then assign it with the passed value.

Comment (rottencandy) [reply webui-src/app/files/files_util.js:131]: The rswebui function is undefined here because there is a circular import happening. The rswebui file is importing a files_util function which is causing files_util to initialize before rswebui does.

Comment (zelfroster) [reply webui-src/app/files/files_util.js:131]: > The rswebui function is undefined here because there is a circular import happening. The rswebui file is importing a files_util function which is causing files_util to initialize before rswebui does.

What to do here?

Comment (rottencandy) [reply webui-src/app/files/files_util.js:131]: Multiple options, you could maybe create a new files_proxy.js to move all the proxyObj related stuff there, so imports aren't circular. Or you could generalize the create proxy util and move it to rswebui itself

Comment (zelfroster) [reply webui-src/app/files/files_util.js:131]: > Multiple options, you could maybe create a new files_proxy.js to move all the proxyObj related stuff there, so imports aren't circular. Or you could generalize the create proxy util and move it to rswebui itself

I have tried both, but the circular import is happening in either ways. I am pushing the code, please guide me.

Comment (zelfroster) [reply webui-src/app/files/files_util.js]: > Multiple options, you could maybe create a new files_proxy.js to move all the proxyObj related stuff there, so imports aren't circular. Or you could generalize the create proxy util and move it to rswebui itself

I have got an idea, I was thinking of exposing eventQueue object in rswebui.js, but seems like events property of the object is already exported, So I am going to try to use it in the files_util.js itself.

Comment (zelfroster) [reply webui-src/app/files/files_util.js]: Alright, It works now. No circular imports. Please review @rottencandy

**Review by rottencandy — APPROVED** [review]
Thanks, looks good.
